File: 33. Search in Rotated Sorted Array.py
class Solution(object):
    def search(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        temp = nums[0]
        if target == temp:
            return 0
        elif target > temp:
            for i in xrange(len(nums)):
                if nums[i] < temp:
                    return -1
                if nums[i] == target:
                    return i
            return -1
        else:
            for i in xrange(len(nums)-1, -1, -1):
                if nums[i] >= temp:
                    return -1
                if nums[i] == target:
                    return i
            return -1

File: 33. Search in Rotated Sorted Array.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int temp = nums[0];
        if (target == temp) return 0;
        else if (target > temp)
        {
            for (int i = 0; i < nums.size(); i++)
            {
                if (nums[i] < temp) return -1;
                if (target == nums[i]) return i;
            }
        }
        else
        {
            for (int i = nums.size()-1; i >= 0; i--)
            {
                if (nums[i] > temp) return -1;
                if (target == nums[i]) return i;
            }
        }
        return -1;
    }
};
